components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/RESTAPIAdminClient.java

```diff
@@ -26,71 +26,77 @@
+		this.qualifiedName = apiId.getProviderName() + "--" + apiId.getApiName() + ":v" +
+				apiId.getVersion();
+		this.qualifiedNonVersionedWebAppName = apiId.getProviderName() + "--" + apiId.getApiName();
+		restApiAdminStub = new RestApiAdminStub(null, environment.getServerURL() + "RestApiAdmin");
+		setup(restApiAdminStub, environment);
+		this.environment = environment;
+	}
+
 	/**
-	 * Add the WebApp to the gateway
+	 * Adds versioned web app configuration to the gateway
+	 *
 	 * @param builder
 	 * @param tenantDomain
 	 * @throws AxisFault
 	 */
-	public void addApi(APITemplateBuilder builder, String tenantDomain ) throws AxisFault {
-        try {
-            String apiConfig = builder.getConfigStringForTemplate(environment);
-            if (tenantDomain != null && !("").equals(tenantDomain)
-                    && !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
-            	restApiAdminStub.addApiForTenant(apiConfig, tenantDomain);
-            }else {
-            	 restApiAdminStub.addApiFromString(apiConfig);	
-            }
-        } catch (Exception e) {
-            throw new AxisFault("Error while adding new WebApp", e);
-        }
+	public void addVersionedWebApp(APITemplateBuilder builder, String tenantDomain)
+			throws AxisFault {
+		try {
+			String apiConfig = builder.getConfigStringForVersionedWebAppTemplate(environment);
+			if (tenantDomain != null && !("").equals(tenantDomain)
+					&& !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+				restApiAdminStub.addApiForTenant(apiConfig, tenantDomain);
+			} else {
+				restApiAdminStub.addApiFromString(apiConfig);
+			}
+		} catch (Exception e) {
+			throw new AxisFault("Error while adding new WebApp", e);
+		}
 	}
 
 	/**
-	 * Get WebApp from the gateway
+	 * Returns versioned web app configuration from the gateway
+	 *
 	 * @param tenantDomain
 	 * @return
 	 * @throws AxisFault
 	 */
-    public APIData getApi(String tenantDomain) throws AxisFault {
-        try {
-        	APIData apiData;
-        	 if (tenantDomain != null && !("").equals(tenantDomain)
-                     && !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
-        		 apiData = restApiAdminStub.getApiForTenant(qualifiedName,tenantDomain);
-             }else {
-            	 apiData = restApiAdminStub.getApiByName(qualifiedName);	
-             }
-            return (APIData) apiData;
-        } catch (Exception e) {
-            throw new AxisFault("Error while obtaining WebApp information from gateway", e);
-        }
-    }
-
-    /**
-     * Update the WebApp in the Gateway
-     * @param builder
-     * @param tenantDomain
-     * @throws AxisFault
-     */
-	public void updateApi(APITemplateBuilder builder, String tenantDomain) throws AxisFault {
+	public APIData getVersionedWebApp(String tenantDomain) throws AxisFault {
+		try {
+			APIData apiData;
+			if (tenantDomain != null && !("").equals(tenantDomain)
```

Comment:
here empty string check is redundant because, when `tenantDomain` equals to `MultitenantConstants.SUPER_TENANT_DOMAIN_NAME` it cannot be empty.
fixed in commit 3a66f276d2d01cc5105007951f897fc77bfbc579 of PR #153